sat/cli/jobstat/main.py

```diff
@@ -0,0 +1,72 @@
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+The main entry point for the jobstat subcommand.
+"""
+import logging
+from sat.apiclient.gateway import APIError, APIGatewayClient
+from sat.report import Report
+from sat.config import get_config_value
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def unload_json(url):
+    """Unloads the JSON object from the server url and formats the data as a dictionary"""
+    try:
+        connect = APIGatewayClient()
+        return connect.get(['statechecker', 'jobstat', 'all']).json()
+    except APIError as err:
+        raise APIError(f'Failed to get BOS session templates: {err}')
+    except ValueError as err:
+        raise APIError(f'Failed to parse JSON in response from BOS when '
+                       f'getting session templates: {err}')
+
+
+def do_jobstat(args):
+    """Run the jobstat command with the given arguments.
+    Args:
+        args: The argparse.Namespace object containing the parsed arguments
+            passed to this subcommand.
+    """
+    try:
+        server_data = unload_json(URL)
+        application_data = server_data.get('jobstat')
+        report = Report(
+            tuple(application_data[0].keys()),
+            None,
+            args.sort_by,
+            args.reverse,
+            get_config_value('format.no_headings'),
+            get_config_value('format.no_borders'),
+            filter_strs=args.filter_strs,
+            display_headings=args.fields,
+            print_format=args.format,
+        )
+        for row in application_data:
+            report.add_row(row)
+        print(report)
+    except Exception as err:
```

Comment:
I would prefer that we don't catch the bare exception class here.

We should most likely only need to catch HTTP errors coming from the API. This comment is also addressed in the gist I left: https://gist.github.hpe.com/eli-kamin/3675a843e0b343f3360f2bcd11327da1